src/system/SystemClock.h

```diff
@@ -150,14 +153,33 @@ class ClockBase
      * Although some platforms may choose to return a value that measures the time since boot for the
      * system, applications must *not* rely on this.
      */
-    Timestamp GetMonotonicTimestamp() { return GetMonotonicMilliseconds64(); }
+    Timestamp GetMonotonicTimestamp()
+    {
+        Timestamp prevTimestamp = mLastTimestamp.load();
+
+        // Force a memory barrier so that the new timestamp is always acquired after any ongoing prevTimestamp store from
+        // other contexts.
```

Comment:
I tried to figure out what this comment is saying and what the fence is guarding against, and failed.

It might be worth writing out at length exactly what this is trying to prevent.....

Reply:
Yeah, will do.

Reply:
Done, PTAL

Reply:
Yes, thank you, that is much clearer!